packages/ckeditor5-image/src/imageupload/imageuploadediting.ts

```diff
@@ -217,14 +256,39 @@ export default class ImageUploadEditing extends Plugin {
 						const loader = fileRepository.loaders.get( uploadId );
 
 						if ( !loader ) {
+							// If the loader does not exist, it means that the image was already uploaded
+							// and the loader promise was removed from the registry. In that scenario we need
+							// to restore response object from the internal map.
+							if ( !isInsertedInGraveyard && this._uploadedImages.has( uploadId ) ) {
+								// Fire `uploadComplete` to set proper attributes on the image element.
+								this.fire<ImageUploadCompleteEvent>( 'uploadComplete', {
+									data: this._uploadedImages.get( uploadId )!,
+									imageElement: imageElement as Element
+								} );
```

Comment:
Maybe we should wrap this in a non-undoable batch as in another place when we fire this event?
```
model.enqueueChange( { isUndoable: false }, writer => {
```

Reply:
I'll check that

Reply:
@niegowski I checked that, works correctly. https://github.com/ckeditor/ckeditor5/pull/17327/commits/1844484b270dbd2c78ba069f65ab54583b559d26